README.md

```diff
@@ -30,6 +30,217 @@ you're a tool author you may be interested in the formal [EBNF grammar][].
 [EBNF grammar]: https://github.com/projectfluent/fluent/tree/master/spec
 
 
+Installation
+------------
+
+    pip install fluent
+
+Usage
+-----
+
+Fluent uses unicode everywhere internally, and doesn't accept bytestrings, so
+that you avoid problems down the road when non-ASCII characters start appearing
+in FTL files. If you are using Python 2, for the following examples you will
+first need to do:
+
+    >>> from __future__ import unicode_literals
+
+or add unicode literal markers to strings.
+
+To generate translations from this Python libary, you start with the
+`MessageContext` class:
+
+    >>> from fluent.context import MessageContext
+
+You pass a list of locales to the constructor - the first being the desired
+locale, with fallbacks after that:
+
+    >>> context = MessageContext(["en-US"], use_isolating=False)
+
+Here we have passed `use_isolating=False` which disables the use of Unicode bidi
+isolation characters, to make the example output easier to read. If you might
+have mixed right-to-left and left-to-right output from your messages then you
+should omit this parameter to get the default `True` value.
+
+You must then add messages. These would normally come from a `.ftl` file stored
+on disk, here we will just add them directly:
+
+    >>> context.add_messages("""
+    ... welcome = Welcome to this great app!
+    ... greet-by-name = Hello, { $name }!
+    ... """)
+
+To generate translations, use the `format` method, passing a message ID and an
+optional dictionary of substitution parameters. If the the message ID is not
+found, a `LookupError` is raised. Otherwise, as per the Fluent philosophy, the
+implementation tries hard to generate *something* even if there are errors. The
+`format` method therefore returns a tuple containing `(translated string,
+errors)`, as below.
+
+    >>> translated, errs = context.format('welcome')
+    >>> translated
+    "Welcome to this great app!"
+    >>> errs
+    []
+
+    >>> translated, errs = context.format('greet-by-name', {'name': 'Jane'})
+    >>> translated
+    "Hello, Jane!"
+
+    >>> translated, errs = context.format('greet-by-name', {})
+    >>> translated
+    "Hello, name!"
+    >>> errs
+    [FluentReferenceError('Unknown external: name')]
+
+
+Numbers
+-------
+
+When rendering translations, Fluent passes any numeric arguments (int or float)
+through locale-aware formatting functions:
+
+    >>> context.add_messages("show-total-points = You have { $points } points.")
+    >>> val, errs = context.format("show-total-points", {'points': 1234567})
+    >>> val
+    'You have 1,234,567 points.'
+
+
+You can specify your own formatting options on the arguments passed in by
+wrapping your numeric arguments with `fluent.types.fluent_number`:
+
+    >>> from fluent.types import fluent_number
+    >>> points = fluent_number(1234567, useGrouping=False)
```

Comment:
`fluent_number` is actually a function - https://github.com/projectfluent/python-fluent/pull/67/files#diff-3e6998f08fb5d73e928f7295d4e2cfa5R205 - it normally returns an instance of `FluentNumber` which is indeed a class. It is just a small wrapper so that we can avoid creating instances unnecessarily. It can also return `FluentNone` instances if you pass `None` (I'm not sure now why or if that is necessary, I may have been following the JS implementation).

We could do clever things inside `FluentNumber.__new__` to avoid creating new instances unnecessarily, but it seems more clever than necessary, and with the `FluentNone` complication